app/src/main/java/org/stepic/droid/ui/util/SearchViewHelper.kt

```diff
@@ -0,0 +1,62 @@
+package org.stepic.droid.ui.util
+
+import android.support.v7.widget.SearchView
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import org.stepic.droid.analytic.Analytic
+import org.stepic.droid.storage.operations.DatabaseFacade
+import org.stepic.droid.ui.custom.AutoCompleteSearchView
+import org.stepic.droid.web.Api
+import java.util.concurrent.TimeUnit
+
+object SearchViewHelper {
+    private const val AUTOCOMPLETE_DEBOUNCE_MS = 300L
+    private const val DB_ELEMENTS_COUNT = 2
+    
+    fun setupSearchViewSuggestionsSources(
+            searchView: AutoCompleteSearchView,
+            api: Api,
+            databaseFacade: DatabaseFacade,
+            analytic: Analytic,
+            onQueryTextSubmit: (() -> Unit)?): CompositeDisposable {
```

Comment:
I think this method with a lot of dependency is strange. Can we use Presenter, scoped to using in SearchView. Dependency will be set in constructor.

Creation of publisher and disposable will be moved to attachView

view will invoke some method for publish query text changed.

So, schedulers will be provided via constructor and code becomes testable. Now hardcoded schedulers is not testable